common/lcs/src/de.rs

```diff
@@ -132,8 +133,27 @@ impl<'de> Deserializer<'de> {
         Ok(u128::from_le_bytes(le_bytes))
     }
 
+    fn parse_u32_from_uleb128(&mut self) -> Result<u32> {
+        let mut value: u64 = 0;
+        for shift in (0..32).step_by(7) {
+            let byte = self.next()?;
+            let num = byte & 0x7f;
+            value |= u64::from(num) << shift;
+            if num == byte {
+                if shift > 0 && num == 0 {
+                    // ULEB128 encoding must be canonical.
+                    return Err(Error::IncorrectUleb128);
+                }
+                // Decoded integer must not overflow.
+                return u32::try_from(value).map_err(|_| Error::IncorrectUleb128);
+            }
+        }
+        // Decoded integer must not overflow.
+        Err(Error::IncorrectUleb128)
+    }
```

Comment:
```suggestion
    fn parse_u32_from_uleb128(&mut self) -> Result<u32> {
    let mut value: u32 = 0;
    let mut shift = 0;
    let mut byte = self.next()? as u32;
    while byte & 0x80 != 0 && shift <= 21 {
        value |= (byte & 0x7f) << shift;
        shift += 7;
        byte = self.next()? as u32;
    }
    if byte == 0 && shift > 0 {
        return Err(Error::Uleb128NonCanonical);
    }
    if byte & 0x80 != 0 || shift > byte.leading_zeros() {
        return Err(Error::Uleb128Overflow);
    }
    value |= byte << shift;
    return Ok(value);
    }
```

Reply:
I'm suggesting some changes which make the code more readable in my opinion (by going from 3 levels of nesting to 1), but this is also extremely subjective, so please feel free to ignore :-)

Reply:
I personally prefer to avoid numbers like `21` when possible, so I'm going to add comment for the test `num == byte` and call it a day.